### PR TITLE
Add channel objects contract testing

### PR DESCRIPTION
**Pull request by parfeon**
test(objects): add channel objects contract testing

Add test case for channel metadata objects contract testing.

### CODE REVIEW

File: Tests/PubNubContractTest/Steps/Objects/PubNubObjectsChannelMetadataContractTestSteps.swift [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+  public override func handleBeforeHook() {
+    channelMetadata = nil
+    super.handleBeforeHook()
+  }
+  
+  public override func setup() {
+    startCucumberHookEventsListening()
+    
+    Given("^the id for '(.*)' channel$") { args, _ in
+      guard args?.count == 1, let channelName = args?.first?.lowercased() else {
```
Comment (jguz-pubnub) [inline Tests/PubNubContractTest/Steps/Objects/PubNubObjectsChannelMetadataContractTestSteps.swift:45]: This is ok, but you can also use the `XCTUnwrap` assert if you would like to:

https://developer.apple.com/documentation/xctest/3380195-xctunwrap
https://www.hackingwithswift.com/example-code/testing/how-to-check-and-unwrap-optionals-in-tests-using-xctunwrap

File: Tests/PubNubContractTest/Steps/Objects/PubNubObjectsTestHelpers.swift [hunk 2]
```diff
@@ -52,31 +52,31 @@ extension PubNubContractTestCase {
...
+        return uuidMetadata
     }
     
-    /// Retrieve `channel` object information using owner name.
+    /// Retrieve `channel metadata` object information using owner name.
     ///
     /// - Parameter name: Entity name which is the same as name of file in which it is stored.
     /// - Returns: Parsed `PubNubChannelMetadata` object or `nil` in case of parse / load error.
-    func channelWithName(_ name: String) -> PubNubChannelMetadata? {
+    func channelMetadataWithName(_ name: String) -> PubNubChannelMetadata? {
```
[the remark this answers is not in the document]
Comment (parfeon) [reply Tests/PubNubContractTest/Steps/Objects/PubNubObjectsTestHelpers.swift:73]: Ah, yep. Will do now.

[the remark this answers is not in the document]
Comment (jguz-pubnub) [reply Tests/PubNubContractTest/Steps/Objects/PubNubObjectsTestHelpers.swift:73]: Perhaps you can do the same for the method above:

```
func uuidMetadataWithName(_ name: String) -> PubNubUUIDMetadata? 
```

File: Tests/PubNubContractTest/Steps/Objects/PubNubObjectsChannelMetadataContractTestSteps.swift (hunk 1 above)
Comment (parfeon) [reply Tests/PubNubContractTest/Steps/Objects/PubNubObjectsChannelMetadataContractTestSteps.swift]: @jguz-pubnub I've pushed changes for this.

File: Tests/PubNubContractTest/Steps/Objects/PubNubObjectsTestHelpers.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (parfeon) [reply Tests/PubNubContractTest/Steps/Objects/PubNubObjectsTestHelpers.swift]: @jguz-pubnub I've pushed changes for this.

File: Tests/PubNubContractTest/Steps/Objects/PubNubObjectsChannelMetadataContractTestSteps.swift (hunk 1 above)
Comment (parfeon) [reply Tests/PubNubContractTest/Steps/Objects/PubNubObjectsChannelMetadataContractTestSteps.swift:45]: @jguz-pubnub had to revert. This won't work properly with `C` functions (`Given` is `C` function defined in `Cucumberish`).

File: Tests/PubNubContractTest/Steps/Objects/PubNubObjectsChannelMetadataContractTestSteps.swift [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+      }
+      
+      self.channelMetadata = channelMetadata
+    }
+    
+    When("^I get the channel metadata(.*)$") { args, _ in
+      let fetchChannelMetadataExpect = self.expectation(description: "Fetch channel metadata response")
+      var includeCustom = false
+      
+      guard let channelMetadataId = self.channelMetadata?.metadataId else { return }
```
Comment (CraigLn) [inline Tests/PubNubContractTest/Steps/Objects/PubNubObjectsChannelMetadataContractTestSteps.swift:77]: Do we want to make an assertion here?

**Review by CraigLn — APPROVED** [review]

**Review by jguz-pubnub — APPROVED** [review]

File: Tests/PubNubContractTest/Steps/Objects/PubNubObjectsChannelMetadataContractTestSteps.swift (hunk 3 above)
Comment (parfeon) [reply Tests/PubNubContractTest/Steps/Objects/PubNubObjectsChannelMetadataContractTestSteps.swift:77]: @CraigLn this one goes after [this](https://github.com/pubnub/swift/pull/118/files/20ecc010f24ad9c6e3b11f67be3d37728c66fd10#diff-4f7451017693fb9dfd82099613a2239b52b80f6f235bf4f3dc13d8692343d37bR44) step and it already have check on `metadataId`

**Pull request merged**